src/numpy-stubs/_typing/_ufunc.pyi

```diff
@@ -380,6 +446,113 @@ class _Call21Bool(Protocol):
         **kwds: Unpack[_Kwargs3],
     ) -> np.bool | NDArray[np.bool]: ...
 
+@type_check_only
+class _Call21Logical(Protocol):
+    @overload  # (scalar, scalar, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (scalar, scalar) -> bool
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.bool: ...
+    @overload  # (array-like, array, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (array-like, array, dtype: dtype[T]) -> Array[T]
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array, array-like, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _AnyArray,
+        x2: _ArrayLikeNumber_co,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (array, array-like, dtype: dtype[T]) -> Array[T]
+    def __call__(
+        self,
+        x1: _AnyArray,
+        x2: _ArrayLikeNumber_co,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array-like, array) -> Array[?]
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
+        /,
+        out: _Out1[_AnyArray] | None = None,
+        *,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array, array-like) -> Array[?]
+    def __call__(
+        self,
+        x1: _AnyArray,
+        x2: _ArrayLikeNumber_co,
+        /,
+        out: _Out1[_AnyArray] | None = None,
+        *,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> NDArray[np.bool]: ...
+    @overload  # (array-like, array-like, out: T) -> T
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: ArrayLike,
+        /,
+        out: _ArrayT | tuple[_ArrayT],
+        *,
+        dtype: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> _ArrayT: ...
```

Comment:
`x2` can also be a scalar or array-like here, so `_ArrayLikeNumber_co` is better.
Also don't forget that array-likes of `object_` are also allowed as input.

Reply:
`out` can be simplified to `_Out1[_ArrayT]`

Reply:
This overload should be placed above the ones with `out: {something that isn't None}`, because this overload has a more specific return type.